microcosm_elasticsearch/factories.py

```diff
@@ -32,26 +32,44 @@ def configure_elasticsearch_client(graph):
     :returns: an Elasticsearch client instance of the configured name
 
     """
+    config = dict()
+
     if strtobool(graph.config.elasticsearch_client.use_aws4auth):
-        kwargs = _configure_aws4auth(graph)
+        configure_elasticsearch_aws(config, graph)
     else:
-        kwargs = dict(
-            hosts=[
-                graph.config.elasticsearch_client.host,
-            ],
-        )
-        if graph.config.elasticsearch_client.username and graph.config.elasticsearch_client.password:
-            kwargs["http_auth"] = (
-                graph.config.elasticsearch_client.username,
-                graph.config.elasticsearch_client.password,
-            )
+        configure_elasticsearch(config, graph)
 
     if strtobool(graph.config.elasticsearch_client.use_python2_serializer):
-        kwargs.update(dict(
+        config.update(
             serializer=JSONSerializerPython2(),
-        ))
+        )
+
+    return Elasticsearch(**config)
+
+
+def configure_elasticsearch(config, graph):
+    """
+    Configure non-AWS elasticsearch
 
-    return Elasticsearch(**kwargs)
+    """
+    config.update(
+        hosts=[
+            graph.config.elasticsearch_client.host,
+        ],
+    )
+
+    if graph.config.elasticsearch_client.username and graph.config.elasticsearch_client.password:
+        config.update(
+            http_auth=(
+                graph.config.elasticsearch_client.username,
+                graph.config.elasticsearch_client.password,
+            ),
+        )
+    elif graph.metadata.testing:
```

Comment:
This is new (default unit test credentials); the rest is just refactoring for clarity